### DISCUSSION TITLE
Implementing UV passbands

### DISCUSSION THREAD

**Discussion by maudegull** [phoebe-project/phoebe2#841] [Implementation]
Hi all,

I have been trying to create an UV passbands. While making it is not an issue I noticed that it seems to not result in bright enough lightcurves when I actually try to model them. Is it possible that PHOEBE clips the wavelength range somewhere? (I tried both ck2004 and blackbody curves). 

When I plot the Inorm() curves for diagnostic it behaves as it should. 
I have been comparing my filter to Johnson:U and the Johnson:U filter is brighter when it should be actually slightly fainter based on the Inorm() curves. Might be that I am missing something specific for the UV filters? 

Thanks a lot for any input or advice 

Best, 

Maude

**Comment by Raindogjones** [comment]
Hi Maude,
This is a bit difficult to diagnose without having the passbands and/or a script that reproduces the issue, but I would suspect this might be a `pblum` issue. Can you check what pblum_mode you are using for your datasets? If you want to directly compare between your filter and Johnson:U, it must be set to 'absolute'.

**Comment by Raindogjones** [comment]
For more information about passband luminosities, see [here](http://phoebe-project.org/docs/latest/tutorials/pblum)

**Reply by maudegull** [reply]
Hi, 

Thanks for the quick reply.
Yeah I have them all set to 'absolute'.
This is how I generate the light curves:

    b = phoebe.default_binary(contact_binary=True)
    SMAXIS = (((PER*u.day)**2*c.G*(M1+M1*Q)*u.Msun/(4*np.pi**2))**(1/3)).to(u.Rsun)
    b.add_dataset('lc', 
          compute_phases=ct,
          dataset='lc_model1',
        passband = 'HST:F336W')
    b.add_dataset('lc', 
          compute_phases=ct,
          dataset='lc_model2',
        passband = 'Johnson:U')
    b.set_value('requiv@primary@component',RAD)
    b.set_value('teff@primary@component',TEFF)
    b.set_value('incl@binary@component',INC)
    b.set_value('period@binary@component',PER)
    b.set_value('sma@binary@component',SMAXIS)
    b.set_value('q@binary@component',Q)
    b.set_value('irrad_frac_refl_bol@primary@component',1.0)
    b.set_value('irrad_frac_refl_bol@secondary@component',1.0)
    b.set_value('gravb_bol@primary@component',1.0)
    b.set_value('gravb_bol@secondary@component',1.0)
    b.set_value('atm@primary@phoebe01@compute','blackbody')
    b.set_value('atm@secondary@phoebe01@compute','blackbody')
    b.set_value('teff@secondary@component',TEFF2)
    b.set_value('t0_supconj@binary@component',tt0)
    b.set_value('pblum_mode', dataset='lc_model1', value='absolute')
    b.set_value('pblum_mode', dataset='lc_model2', value='absolute')
    b.add_compute(phoebe.compute.phoebe,dataset= ['lc_model1','lc_model2'] ,compute='preview',irrad_method='none',overwrite=True)
    b.run_compute(compute='preview')
    times = b.get_parameter('compute_phases', dataset='lc_model1').get_value()
    F336W = b.get_parameter('fluxes',context='model', dataset='lc_model1').get_value()
    JOHNSON = b.get_parameter('fluxes',context='model', dataset='lc_model2').get_value()

Where I have tried both 'blackbody' and 'ck2004. 
It works for all my other custom passbands, except for the UV ones.

**Reply by Raindogjones** [reply]
Nothing looks obviously wrong there to me.
Do you set the limb-darkening anywhere?

**Reply by maudegull** [reply]
I'm not specifying it anywhere, I'm using the default setting, which is 'lookup' i think? 
I also added the passband to this [link](https://drive.google.com/drive/folders/1BqplaQlJhoTg8iDNizh0LgBf5nPUzMiX?usp=drive_link).

**Reply by maudegull** [reply]
![Download - 2024-04-03T144931 501](https://github.com/phoebe-project/phoebe2/assets/19806342/18f8a408-acc1-42c0-a29e-cc86eaececf0)

Here is also the Inorm plot which looked ok to me on first glance.

**Reply by Raindogjones** [reply]
Yeah, looks fine.  Can you see what happens when you turn limb-darkening off?
b.set_value_all('ld_mode','manual')
b.set_value_all('ld_func','linear')
b.set_value_all('ld_coeffs',0)

The difference in integrated flux could be due to steeper limb-darkening in your bluer passband (i.e., Imu rather than Inorm).

**Reply by maudegull** [reply]
Hi David, 
Tried that too and it doesn't make a difference.
That's why I thought maybe the blue side of F336W is getting clipped somewhere when computing the integrated flux?

(Also FYI possible that the passband I just sent the link to only has the 'ck2004' computed)

**Reply by Raindogjones** [reply]
Once the passband file is created, the colour doesn't really matter any more. Phoebe just interpolates the tables of Inorm and Imu to find the emergent intensity of each surface triangle and then integrates over them. So, if Inorm is sensible then there's no way to clip the blue anywhere else...Once the passband file is created, the colour doesn't really matter.  Phoebe just interpolates Inorm (or Imu) to find the emergent intensity of each triangle and then integrates over them, so no way for it to be blue-clipped after creating the passband file.  The gdrive link isn't public (I've requested access but you probably need to go and grant it).

**Reply by maudegull** [reply]
Makes sense. 
I updated the sharing, so hopefully it works now?

**Reply by Raindogjones** [reply]
Yep, I'm playing with it now to see if I can reproduce your issue (but I'm on shift at the telescope so it might take a while).

**Reply by Raindogjones** [reply]
I've replicated the issue but I'm a bit stuck as to what exactly is going on.  Maybe @aprsa or @kecnry have an idea?

**Reply by maudegull** [reply]
Thanks for looking into this!

**Comment by aprsa** [comment]
Castelli & Kurucz atmospheres are in the 90nm-40um range, so if your passband is bluer than that, there will indeed be issues. For blackbody, on the other hand, there should be no issues with wavelength. I'll investigate this pm or tomorrow and get back to you.

**Reply by maudegull** [reply]
Hi Andrej, 

Thanks for replying so quickly and looking into it.
Just some more info on the passband.
F336W central wavelength is at 336 nm, and the actual wavelength range of the PTF file I use ranges from 100nm-30um. 
I also have tried clipping the ptf closer to the actual transmission curve, but that didn't really make a difference either.

**Comment by aprsa** [comment]
@maudegull, could you please point me to the actual passband transmission function that you're trying to support?

**Reply by maudegull** [reply]
Yeah np!

import pysynphot as S
f336w_wfc3 = S.ObsBandpass('wfc3,uvis2,f336w')

Thanks!

**Reply by aprsa** [reply]
this is what I'm getting, does that look right? (I'm only taking throughputs > 1e-5)
![f336w](https://github.com/phoebe-project/phoebe2/assets/2669162/68171d1d-fdbb-419c-bda7-0bb1f1dbe569)

**Reply by maudegull** [reply]
Yepp, that looks as expected!

**Reply by aprsa** [reply]
This is what I'm getting for Inorm comparison:
![inorm](https://github.com/phoebe-project/phoebe2/assets/2669162/bf69ef9d-47a1-4ce1-95aa-dd2c7a1ed705)
And this is what I'm getting for the default binary:
![comp](https://github.com/phoebe-project/phoebe2/assets/2669162/9770c17c-323e-4700-88f6-0ec325586633)
Are you seeing the same thing?

**Reply by maudegull** [reply]
Yepp, I am seeing the same thing for the Inorm plot, and for the phoebe default binary.

**Reply by aprsa** [reply]
@maudegull, please forgive the delay in getting back to this -- total solar eclipse and the related activities got in the way and I'm only getting back to this now.

So, I believe that what you are seeing is correct: in absolute units, the system in Johnson U is indeed brighter than in F336W because Johnson U is significantly broader (about 6x larger area), thus letting more light pass through. When you plot Inorm as a function of temperature, these values are passband-normalized, i.e. they're divided by ptf_area=\int P(\lambda) d\lambda, that's why F336W appears to be brighter. But when computing actual luminosities, we need to multiply by ptf_area, which causes the observed behavior where Johnson U light curve in absolute units is brighter. You can access passband areas as:
```
>>> f336w = phoebe.get_passband('HST:f336w')
>>> johnson_u = phoebe.get_passband('Johnson:U')
>>> print(f'{f336w.ptf_photon_area=}, {johson_u.ptf_photon_area=})  # for photon-counting
f336w.ptf_photon_area=3.51775166077845e-15, johnson_u.ptf_photon_area=2.10524873361628e-14
>>> print(f'{f336w.ptf_photon_area=}, {johson_u.ptf_photon_area=})  # for energy-weighted
f336w.ptf_area=1.04746375753476e-08, johnson_u.ptf_area=5.85188917293367e-08
```
Photon-counting area ratio is thus ~6, and the flux ratio between the two light curves needs to be around 6 as well (accounting for limb darkening, reflection, etc, as well):
![image](https://github.com/phoebe-project/phoebe2/assets/2669162/9a360c2a-125d-41e3-922e-9f7925923c3a)
Thus, I believe it makes sense what you're seeing here. Does this address your concern?

**Reply by maudegull** [reply]
Hi Andrej, 

No problem at all!

I see what you are saying, but I’m still a little confused as what is array.

To my understanding in the W/m**2 plot here F275W and F336W should be brighter which they are not. Correct?

<img width="553" alt="Bildschirmfoto 2024-04-15 um 13 51 13" src="https://github.com/phoebe-project/phoebe2/assets/19806342/d6b393a6-0341-4777-b532-26bd1b3da042">

The light curves here are generated by setting 'pblum_mode' to 'absolute'. 
and this is returned when i type:
    print (b.get_parameter('fluxes',context='model', dataset='F336W'))

Parameter: fluxes@F336W@latest@model
                       Qualifier: fluxes
                     Description: Model (synthetic) flux
                           Value: [1.65537110e+27 1.66336310e+27 1.68624242e+27 ... 1.68385638e+27
 1.66203247e+27 1.65537110e+27] W / m2
                  Constrained by: 
                      Constrains: None
                      Related to: None


For my observational data converting my magnitudes to W/m**2 works for all the bands except for F275W and F336W where they do not match, they are too faint. 

Here is an example of what I am talking about with F475W for reference

<img width="567" alt="Bildschirmfoto 2024-04-15 um 13 50 51" src="https://github.com/phoebe-project/phoebe2/assets/19806342/780a4311-5e2f-44b4-982e-d7c3db59e2e2">

**Reply by maudegull** [reply]
Sorry to clarify, 
What I am understanding from your response is that in units of W indeed John U should be brighter because the area over which the flux was collected is larger, but in the units of W/m**2, then F275W and F336W should be brighter. 

But maybe I am misinterpreting, please let me know if that's not it.

**Reply by aprsa** [reply]
> To my understanding in the W/m**2 plot here F275W and F336W should be brighter which they are not. Correct?

I don't think so, no. Flux in a certain passband (measured in W/m^2) depends on the SED intensity within the passband (W/m^3) integrated over that passband. Thus, how "bright" an object appears depends on how much energy is radiated in that passband and how transmissive is the passband. Given that Johnson U is much wider than F275W or F336W, I would indeed expect that Johnson U fluxes are higher. If, say, SED intentities under Johnson U were ~6 times smaller than SED intensities in F336W, I would expect the two curves to be roughly aligned.

> What I am understanding from your response is that in units of W indeed John U should be brighter because the area over which the flux was collected is larger, but in the units of W/m**2, then F275W and F336W should be brighter.

That doesn't sound right to me either: while yes, passband luminosity (measured in W) in Johnson U should be larger than passband luminosity of F336W, so should the fluxes -- m^2 comes from 4 \pi D^2, where D is distance. That's the same for both passbands, so if passband luminosity is greater, so is the flux.

Here is a [notebook](http://phoebe-project.org/workshops/2023june/Tutorial_flux_calibration) that I prepared for the workshop last year that discusses some of these concepts, perhaps it will be useful for this discussion.

I will try to write up a small example that demonstrates this better at some point next week; I'm swamped right now but I think this is a very useful discussion that merits elaboration so I'll commit some time to write it up. If you'd like, we can try to use this specific example to demonstrate how phoebe's radiative quantities work under the hood.

**Reply by maudegull** [reply]
Hi Andrej, 

Thanks for the clarification
Okay so I originally thought the m^2 based on definitions would come from the D^2, but since my conversions were off.
The notebook was helpful too! I might just not have been diligently enough in my search for it. 
I think that would be helpful for someone using PHOEBE from the first time!
I think what was really getting me confused is really the scaling within PHOEBE that happens between the different modes and how to relate that to actual observations. 
I'm still slightly off in my conversions; to my understanding, this is what should happen to get to magnitude. 
If set to absolute, Phoebe will spit out absolute flux with units W/m^2.  I then need to divide the value by the ptf_photon_area. This leaves me with W/m^2/m. If I divide this by 1e7 I should get it in units erg/s/cm^2/A. Then I can use the Vega_zero_flux to get to Vega system before converting it into the VEGA magnitude.
Am I missing something here? 
A quick tutorial on how to get observed magnitudes to something you can compare to the PHOEBE light curves would be great! 

Thanks a lot for your help and patienece!

**Reply by maudegull** [reply]
Hi, 

I just wanted to follow-up real qiuck on the conversion. This conversion scales my magnitudes to Phoebes output fluxes:

flux275 = zeropointflux x 10**(-0.4 x magnitude) x 1e7(_erg/s/cm^2/A -> _W/m^3_) x (distance)**2 / (centralwavelength in meters) x pf275w.ptfphotonarea

To be honest the last chunk (i.e. centralwavelength and ptfphotonarea) is the part that I am most unsure about, but without that scaling I just can't get the converted magnitudes to be matching with the Phoebe output.
